app/src/renderer/vuex/modules/send.js

```diff
@@ -60,44 +73,69 @@ export default ({ node }) => {
 
       const requestMetaData = {
         sequence: state.nonce,
-        name: `anonymous`,
         from: rootState.wallet.address,
-        account_number: rootState.wallet.accountNumber, // TODO move into LCD?
+        account_number: rootState.wallet.accountNumber,
         chain_id: rootState.connection.lastHeader.chain_id,
-        gas: String(config.default_gas),
+        gas: String(state.externals.config.default_gas),
         generate_only: true
       }
       args.base_req = requestMetaData
 
       // extract type
-      const type = args.type || `send`
+      const type = args.type || `send` // TODO: why send as default ??
       delete args.type
 
+      // get signing method: localkeystore or ledger
+      const submitType = args.submitType || `local`
+      delete args.submitType
+
       // extract "to" address
       const to = args.to
       delete args.to
 
       // get the generated tx by querying it from the backend
       const req = to ? node[type](to, args) : node[type](args)
       const generationRes = await req.catch(handleSDKError)
-
-      // get private key to sign
-      const wallet = getKey(rootState.user.account, args.password)
-      delete args.password
-
-      // sign
       const tx = generationRes.value
-      const signature = sign(tx, wallet, requestMetaData)
 
-      // broadcast
-      const signedTx = createSignedTx(tx, signature)
-      const body = createBroadcastBody(signedTx)
+      let signature
+      if (rootState.ledger.isConnected && submitType === `ledger`) {
+        // TODO: move to wallet script
+        const signMessage = state.externals.createSignMessage(
+          tx,
+          requestMetaData
+        )
+        const signatureByteArray = await dispatch(`signWithLedger`, signMessage)
+        // we have to parse the signature from Ledger as it's in DER format
+        const signatureBuffer = state.externals.signatureImport(
+          signatureByteArray
+        )
+        /* 
+          NOTE: message is not hashed since Ledger needs to display it on screen
+          and then internally hashes the json before signing it
+        */
+        signature = state.externals.createSignature(
+          signatureBuffer,
+          requestMetaData.sequence,
+          requestMetaData.account_number,
+          rootState.ledger.pubKey
+        )
+      } else {
```

Comment:
what if `rootState.ledger.isConnected === false && submitType === `ledger`

Reply:
I did that (i.e `(!rootState.ledger.isConnected || submitType === "local")`)  but doesn't make sense until we have the signer app

Reply:
let's test what happens if ledger is disconnected in between.